tests/src/JIT/Directed/StructABI/StructABI.cs

```diff
@@ -646,471 +670,1538 @@ public bool Equals(Nested9 other)
+		TwoLongs nativeEnoughRegistersSysV3 = EnoughRegistersSysV3(1.0d, 2.0d, 3.0d, 4.0d, 5.0d, 6.0d, 7.0d, 8.0d, expectedEnoughRegistersSysV3);
+		TwoLongs managedEnoughRegistersSysV3 = EnoughRegistersSysV3Managed(1.0d, 2.0d, 3.0d, 4.0d, 5.0d, 6.0d, 7.0d, 8.0d, expectedEnoughRegistersSysV3);
+
+		if (!expectedEnoughRegistersSysV3.Equals(nativeEnoughRegistersSysV3))
+		{
+			Console.WriteLine("Native EnoughRegistersSysV3 failed");
+			ok = false;
+		}
+
+		if (!expectedEnoughRegistersSysV3.Equals(managedEnoughRegistersSysV3))
+		{
+			Console.WriteLine("Managed EnoughRegistersSysV3 failed");
+			ok = false;
+		}
+
+		return ok;
+	}
+
+	static bool EnoughRegistersSysV4Wrapper()
+	{
+		bool ok = true;
+
+		DoubleAndByte expectedEnoughRegistersSysV4 = DoubleAndByte.Get();
+		DoubleAndByte nativeEnoughRegistersSysV4 = EnoughRegistersSysV4(1.0d, 2.0d, 3.0d, 4.0d, 5.0d, 6.0d, 7.0d, expectedEnoughRegistersSysV4);
+		DoubleAndByte managedEnoughRegistersSysV4 = EnoughRegistersSysV4Managed(1.0d, 2.0d, 3.0d, 4.0d, 5.0d, 6.0d, 7.0d, expectedEnoughRegistersSysV4);
+
+		if (!expectedEnoughRegistersSysV4.Equals(nativeEnoughRegistersSysV4))
+		{
+			Console.WriteLine("Native EnoughRegistersSysV4 failed");
+			ok = false;
+		}
+
+		if (!expectedEnoughRegistersSysV4.Equals(managedEnoughRegistersSysV4))
+		{
+			Console.WriteLine("Managed EnoughRegistersSysV4 failed");
+			ok = false;
+		}
+
+		return ok;
+	}
+
+	static int Main()
+	{
+		var ok = true;
+	
+		ok = EchoSingleByteWrapper();
+		ok = EchoSingleLongWrapper();
+		ok = EchoSingleFloatWrapper();
+		ok = EchoSingleDoubleWrapper();
+		ok = EchoByteAndFloatWrapper();
+		ok = EchoLongAndFloatWrapper();
+		ok = EchoByteAndDoubleWrapper();
+		ok = EchoDoubleAndByteWrapper();
+		ok = EchoPointerAndByteWrapper();
+		ok = EchoByteAndPointerWrapper();
+		ok = EchoByteFloatAndPointerWrapper();
+		ok = EchoPointerFloatAndByteWrapper();
+		ok = EchoShortIntFloatIntPtrWrapper();
+		ok = EchoTwoLongsWrapper();
+		ok = EchoTwoFloatsWrapper();
+		ok = EchoTwoDoublesWrapper();
+		ok = EchoFourLongsWrapper();
+		ok = EchoFourDoublesWrapper();
+		ok = EchoInlineArray1Wrapper();
+		ok = EchoInlineArray2Wrapper();
+		ok = EchoInlineArray3Wrapper();
+		ok = EchoInlineArray4Wrapper();
+		ok = EchoInlineArray5Wrapper();
+		ok = EchoInlineArray6Wrapper();
+		ok = EchoNested1Wrapper();
+		ok = EchoNested2Wrapper();
+		ok = EchoNested3Wrapper();
+		ok = EchoNested4Wrapper();
+		ok = EchoNested5Wrapper();
+		ok = EchoNested6Wrapper();
+		ok = EchoNested7Wrapper();
+		ok = EchoNested8Wrapper();
+		ok = EchoNested9Wrapper();
+		ok = NotEnoughRegistersSysV1Wrapper();
+		ok = NotEnoughRegistersSysV2Wrapper();
+		ok = NotEnoughRegistersSysV3Wrapper();
+		ok = NotEnoughRegistersSysV4Wrapper();
+		ok = NotEnoughRegistersSysV5Wrapper();
+		ok = NotEnoughRegistersSysV6Wrapper();
+		ok = EnoughRegistersSysV1Wrapper();
+		ok = EnoughRegistersSysV2Wrapper();
+		ok = EnoughRegistersSysV3Wrapper();
+		ok = EnoughRegistersSysV4Wrapper();
+
```

Comment:
This test is going to pass now if the last test passes -- it will ignore all other test case results.

Reply:
Thank you for catching this. I have opened #18529 to address it.